src/uipath/agent/models/agent.py

```diff
@@ -19,6 +19,103 @@ class AgentResourceType(str, Enum):
+    )
+
+
+class AgentGuardrailRule(BaseModel):
+    """Guardrail validation rule."""
+
+    rule_type: str = Field(..., alias="$ruleType")
+    rule_type_alt: Optional[str] = Field(None, alias="ruleType")
+    validator: str = Field(..., description="Validator type")
+    parameters: List[AgentGuardrailRuleParameter] = Field(
+        default_factory=list, description="Rule parameters"
+    )
+
+    model_config = ConfigDict(
+        validate_by_name=True, validate_by_alias=True, extra="allow"
+    )
+
+
+class AgentGuardrailActionApp(BaseModel):
+    """App configuration for guardrail actions."""
+
+    name: str = Field(..., description="App name")
+    version: str = Field(..., description="App version")
+    folder_name: str = Field(..., alias="folderName", description="Folder name")
+
+    model_config = ConfigDict(
+        validate_by_name=True, validate_by_alias=True, extra="allow"
+    )
+
+
+class AgentGuardrailActionRecipient(BaseModel):
+    """Recipient for guardrail actions."""
+
+    type: int = Field(..., description="Recipient type")
+    value: str = Field(..., description="Recipient identifier")
+    display_name: str = Field(..., alias="displayName", description="Display name")
+
+    model_config = ConfigDict(
+        validate_by_name=True, validate_by_alias=True, extra="allow"
+    )
+
+
+class AgentGuardrailAction(BaseModel):
+    """Action configuration for guardrails."""
+
+    action_type: str = Field(..., alias="$actionType")
+    action_type_alt: Optional[str] = Field(None, alias="actionType")
+    app: AgentGuardrailActionApp = Field(..., description="App configuration")
+    recipient: AgentGuardrailActionRecipient = Field(..., description="Recipient")
+
+    model_config = ConfigDict(
+        validate_by_name=True, validate_by_alias=True, extra="allow"
+    )
+
+
+class AgentGuardrailSelector(BaseModel):
+    """Selector for guardrail application scope."""
+
+    scopes: List[str] = Field(..., description="Scopes where guardrail applies")
+    match_names: List[str] = Field(
+        ..., alias="matchNames", description="Names to match"
+    )
+
+    model_config = ConfigDict(
+        validate_by_name=True, validate_by_alias=True, extra="allow"
+    )
+
+
+class AgentGuardrailPolicy(BaseModel):
+    """Complete guardrail policy definition."""
+
+    id: str = Field(..., description="Policy identifier")
+    name: str = Field(..., description="Policy name")
+    description: str = Field(..., description="Policy description")
+    rules: List[AgentGuardrailRule] = Field(..., description="Validation rules")
+    action: AgentGuardrailAction = Field(..., description="Action configuration")
+    enabled_for_evals: bool = Field(
+        ..., alias="enabledForEvals", description="Enabled for evaluations"
+    )
+    selector: AgentGuardrailSelector = Field(..., description="Selector configuration")
+
+    model_config = ConfigDict(
+        validate_by_name=True, validate_by_alias=True, extra="allow"
+    )
+
+
```

Comment:
let's also add a similar test in `tests/agent/models` to make sure the deserialization of an `agent.json` with guardrails works as expected